Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -58,7 +58,7 @@ public enum MacroDefinition {
   /// defining macro. These subtrees will need to be replaced with the text of
   /// the corresponding argument to the macro, which can be accomplished with
   /// `MacroDeclSyntax.expandDefinition`.
-  case expansion(MacroExpansionExprSyntax, replacements: [Replacement])
+  case expansion(MacroExpansionExprSyntax, replacements: [Replacement], genericReplacements: [GenericArgumentReplacement]) // FIXME: do we need to evolve this without breaking? I assume so?
```

Comment:
FIXME: I assume we probably can't change this and need to add a new case since we need to be compatible with various versions of compiler?

(I have no idea if we give such guarantees here)

Reply:
`SwiftSyntaxMacroExpansion` module is intended to be only used by the compiler that's built at the same time. We don't need to worry about the version compatibility as long as `swift` and `swift-syntax` repo are synced when building.

Reply:
It would definitely need to be release-noted though because it’s changing public API and it might make sense to define a compatibility layer to minimize the breakage.

```swift
static func expansion(MacroExpansionExprSyntax) -> MacroDefinition
```

Reply:
Ok, I'll add a func like that -- it'll still break code exhaustively switching over the enum, but if the compiler is expected to revlock that's good.

Reply:
Yes, the static func is just to reduce the number cases where it breaks clients to a minimum.